tool/src/js/conversions.rs

```diff
@@ -54,6 +54,10 @@ pub fn gen_value_js_to_rust(
                 param_name, param_name, align
             ));
         }
+        ast::TypeName::Primitive(PrimitiveType::char) => {
+            pre_logic.push(format!("if (!{p}.length || !{p}.codePointAt || {p}.length != 1) {{ throw new TypeError(\"Expected single-character string for `char` parameter {p}, found \" + {p}); }}", p=param_name));
```

Comment:
Issue: `"😋".length === 2`

Reply:
Fortunately JS string _iteration_ does the sensible thing here